src/Features/Core/Portable/DocumentationComments/CodeFixes/AbstractAddDocCommentNodesCodeFixProvider.cs

```diff
@@ -0,0 +1,173 @@
+                var nodeAfterNewParamNode = GetLastParamNodeCorrespondingToParameterInList(paramNodes.AsEnumerable().Reverse(), paramsAfterCurrentParam) ??
+                                                   newDocComment.ChildNodes().First();
+
+                // Adjust for doc comment marker before the node
+                if (nodeAfterNewParamNode != null)
+                {
+                    var paramNodeSiblings = nodeAfterNewParamNode.Parent.ChildNodes().ToList();
+                    var indexOfNode = paramNodeSiblings.IndexOf(nodeAfterNewParamNode);
+
+                    // set insert node to be the doc comment signifier of the closest param before the new node
+                    if (indexOfNode > 0 && paramNodeSiblings[indexOfNode - 1] is TXmlTextSyntax)
+                    {
+                        nodeAfterNewParamNode = paramNodeSiblings[indexOfNode - 1];
+                    }
+                }
+
+                var newNodeList = new[]
+                {
+                    // the last value will almost always be true, unless the node is embedded in another doc comment node
+                    GetNewNode(parameterName, nodeAfterNewParamNode == newDocComment.ChildNodes().First())
+                };
+                newDocComment = newDocComment.InsertNodesBefore(nodeAfterNewParamNode, newNodeList);
+            }
+
+            var newRoot = root.ReplaceNode(docCommentNode, newDocComment.WithAdditionalAnnotations(Formatter.Annotation));
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private List<TXmlElementSyntax> GetParameterNodes(SyntaxNode docComment, string nodeName)
+        {
+            var nodes = docComment.ChildNodes().OfType<TXmlElementSyntax>()
+                                               .Where(w => GetXmlElementLocalName(w) == nodeName)
+                                               .ToList();
+
+            // Prefer to place the doc comment in the outer level, as in auto-created comments
+            if (!nodes.Any())
+            {
+                nodes = docComment.DescendantNodes(descendIntoChildren: _ => true)
+                                  .OfType<TXmlElementSyntax>()
+                                  .Where(w => GetXmlElementLocalName(w) == nodeName)
+                                  .ToList();
+            }
+
+            return nodes;
+        }
+
+        private bool NodeExists(IEnumerable<TXmlElementSyntax> paramNodes, string name)
+        {
+            return paramNodes.Select(GetNameAttributes)
+                             .Where(nameAttributes => nameAttributes.Count == 1)
+                             .Any(nameAttributes => nameAttributes.Select(GetValueFromNameAttribute).Contains(name));
+        }
+
+        protected TXmlElementSyntax GetLastParamNodeCorrespondingToParameterInList(
+            IEnumerable<TXmlElementSyntax> paramNodeList,
+            List<string> methodParamSubset)
```

Comment:
So i'm still uncertain why you need to pass in a list here.  You're either making a param-node for the first parameter, in which case you would nto call this.  Or you're making a param-node for parameter ```i``` where ```i > 0```.  However, for all of these, you must have already made the parameter for ```i - 1```.  So you just need to pass in the previous param-name and look for the matching param-node for it.